src/content/en/updates/posts/2016/05/houdini.md

```diff
@@ -0,0 +1,366 @@
+implementation for parallax scrolling using the compositor worklet.
+
+{% highlight JS %}
+// main.js
+window.compositorWorklet.import('worklet.js')
+  .then(function() {
+    var animator = new CompositorAnimator('parallax');
+    animator.postMessage([
+      new CompositorProxy($('.scroller'), ['scrollTop']),
+      new CompositorProxy($('.parallax'), ['transform']),
+    ]);
+  });{% endhighlight %}
+{% highlight JS %}
+// worklet.js
+registerCompositorAnimator('parallax', class {
+  tick(timestamp) {
+    var t = self.parallax.transform;
+    t.m42 = -0.1 * self.scroller.scrollTop;
+    self.parallax.transform = t;
+  }
+
+  onmessage(e) {
+    self.scroller = e.data[0];
+    self.parallax = e.data[1];
+  };
+});
+{% endhighlight %}
+
+My colleague Robert Flack has written a [polyfill][CompWorklet polyfill] for the
+compositor worklet so you can give it a try already – obviously with a much
+higher performance impact.
+
+### Layout Worklet ([spec][Layout Worklet spec])
+
+Again, a specification that is practically empty, but the concept is intriguing:
+Write your own layout! The layout worklet is supposed to enable you to do
+`display: layout('myLayout')` and run your JavaScript to arrange a node’s
+children in the node’s box . Of course, running a full JavaScript implementation
+of CSS’s `flex-box` layout will be slower than running an equivalent native
+implementation – but it’s easy to imagine a scenario where cutting corners can
+yield a performance gain. Imagine a website consisting of nothing but tiles á la
+Windows 10 or a [Masonry]-style layout. Absolute/fixed positioning is not used,
+neither is `z-index` nor do elements ever overlap or have any kind of border or
+overflow. Being able to skip all these checks on re-layout could yield a
+performance gain.
+
+{% highlight JS %}
+registerLayout('random-layout', class {
+    static get inputProperties() {
+      return [];
+    }
+    static get childrenInputProperties() {
+      return [];
+    }
+    layout(children, constraintSpace, styleMap) {
+        Const width = constraintSpace.width;
+        Const height =constraintSpace.height;
+        for (let child of children) {
+            const x = Math.random()*width;
+            const y = Math.random()*height;
+            const constraintSubSpace = new ConstraintSpace();
+            constraintSubSpace.width = width-x;
+            constraintSubSpace.height = height-y;
+            const childFragment = child.doLayout(constraintSubSpace);
+            childFragment.x = x;
+            childFragment.y = y;
+        }
+
+        return {
+            minContent: 0,
+            maxContent: 0,
+            width: width,
+            height: height,
+            fragments: [],
+            unPositionedChildren: [],
+            breakToken: null
+        };
+    }
+});
+{% endhighlight %}
+
+### Typed CSSOM ([spec][Typed CSSOM spec])
+That’s enough worklets for now, let’s get back to some good ol’ APIs. Typed
```

Comment:
Well, the APIs so far can only be used inside their respective worklets. This one is just a “normal” main thread API. I might just drop it, yeah.